### PR TITLE
[WIP] Version 2.10

### PR DESCRIPTION
**Pull request by zackkatz**

### CODE REVIEW

**Comment by zackkatz** [comment]
- [ ] Prevent "Are you sure you want to leave this page" message when clicking the Update (submit post) button

**Comment by zackkatz** [comment]
- [ ] Merge Tag autocomplete should support type-ahead ( Start typing `{all_` and the list should filter by searching for `{all_`)
- [ ] When autocomplete list is open, the escape key should close it, but not close the jQuery dialog

**Comment by zackkatz** [comment]
- [ ] Add support for opening websites in Fancybox
- [ ] Add support for opening videos in Fancybox

File: includes/class-admin-installer.php [hunk 1]
```diff
@@ -311,26 +316,47 @@ public function render_screen() {
...
-					$this->render_download( $extension, $wp_plugins );
-				}
+				$this->render_section(  'friends', esc_html__( 'Friends of GravityView', 'gravityview' ), $downloads_data, $wp_plugins );
 				?>
             </div>
         </div>
 		<?php
 	}
 
+	private function render_section( $section_slug, $heading, $downloads_data, $wp_plugins = array() ) {
```
Comment (zackkatz) [inline includes/class-admin-installer.php:334]: Needs docblock

File: includes/extensions/entry-notes/class-gravityview-field-notes.php [hunk 2]
```diff
@@ -99,9 +101,10 @@ public function add_entry_default_field( $entry_default_fields, $form, $zone ) {
 
 		if( in_array( $zone, array( 'directory', 'single' ) ) ) {
 			$entry_default_fields['notes'] = array(
-				'label' => __( 'Entry Notes', 'gravityview' ),
-				'type'  => 'notes',
-				'desc'  => __( 'Display, add, and delete notes for an entry.', 'gravityview' ),
+					'label' => __( 'Entry Notes', 'gravityview' ),
```
Comment (zackkatz) [inline includes/extensions/entry-notes/class-gravityview-field-notes.php:104]: Spacing

File: includes/fields/class-gravityview-field-consent.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			'type'    => 'radio',
+			'class'   => 'vertical',
+			'label'   => __( 'What should be displayed:', 'gravityview' ),
+			'value'   => $default,
+			'desc'    => '',
+			'choices' => array(
+				'both' => __( 'Consent image with description', 'gravityview' ),
+				'tick' => __( 'Consent image', 'gravityview' ),
+				'label' => __( 'Consent description', 'gravityview' ),
+			),
```
Comment (zackkatz) [inline includes/fields/class-gravityview-field-consent.php:116]: `display` group

File: includes/fields/class-gravityview-field-id.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	var $group = 'meta';
+
+	var $icon = 'dashicons-code-standards';
+
+	var $is_numeric = true;
+
+	public function __construct() {
+		$this->label = esc_html__( 'Entry ID', 'gravityview' );
+	    $this->description = __('The unique ID of the entry.', 'gravityview');
```
Comment (zackkatz) [inline includes/fields/class-gravityview-field-id.php:25]: spacing

File: includes/fields/class-gravityview-field-ip.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+	var $group = 'meta';
+
+	var $icon = 'dashicons-laptop';
+
+	var $is_numeric = true;
+
+	public function __construct() {
+		$this->label = __('User IP', 'gravityview');
+	    $this->description = __('The IP Address of the user who created the entry.', 'gravityview');
```
Comment (zackkatz) [inline includes/fields/class-gravityview-field-ip.php:25]: Spacing

File: includes/fields/class-gravityview-field-sequence.php [hunk 6]
```diff
@@ -34,15 +34,47 @@ class GravityView_Field_Sequence extends GravityView_Field {
...
+
+		if ( 'edit' === $zone ) {
+			return $entry_default_fields;
+		}
+
+		$entry_default_fields['sequence'] = array(
+			'label' => __( 'Result Number', 'gravityview' ),
+			'type'  => $this->name,
+			'desc'  => $this->description,
+			'icon'  => $this->icon,
```
Comment (zackkatz) [inline includes/fields/class-gravityview-field-sequence.php:72]: Group `gravityview`

File: includes/presets/register-default-templates.php [hunk 7]
```diff
@@ -39,3 +39,77 @@ function gravityview_register_default_templates() {
...
+			}
+
+			$placeholder['type']     = 'custom';
+			$placeholder['included'] = ( (int) $license['price_id'] >= $placeholder['price_id'] );
+
+			new GravityView_Placeholder_Template( $placeholder['slug'], $placeholder );
+		}
+
+	} catch ( Exception $exception ) {
+
```
Comment (zackkatz) [inline includes/presets/register-default-templates.php:112]: Add logging

File: assets/css/scss/admin-metabox-panel.scss [hunk 8]
```diff
@@ -6,13 +6,20 @@
...
+  width: 100%;
   border-left: 1px solid #eee;
+
+  /**
+   * The sidebar is 30% wide, max-width 20em.
+   * This code below is to grow the content panel, without getting into flexbox
+   */
+  padding-left: 20em;
+  padding-left: calc( min( 30%, 20em ) );
+  margin-left: -20em;
```
Comment (zackkatz) [inline assets/css/scss/admin-metabox-panel.scss:21]: Confirm this works in RTL

File: assets/css/scss/admin-tooltips.scss [hunk 9]
```diff
@@ -16,133 +16,355 @@
 		font: normal normal normal 14px/1 FontAwesome;
 		content: "\f059";
 	}
-
-	margin-left: 0;
+	display: inline-block;
+	text-align: center;
+	border-radius: 100px;
+	padding: .2rem;
+	margin: 0 2px 0 0;
```
Comment (zackkatz) [inline assets/css/scss/admin-tooltips.scss:23]: RTL

**Comment by zackkatz** [comment]
- [ ] Update to latest Fancybox

File: future/includes/class-gv-renderer.php [hunk 10]
```diff
@@ -36,54 +37,119 @@ public static function maybe_print_notices( $gravityview = null ) {
...
+
+		$notice = wpautop( implode( ' ', $message_strings ) );
+
+		echo \GVCommon::generate_notice( $notice, 'warning', 'edit_gravityview', $gravityview->view->ID );
+	}
+
+	/**
+	 * Check empty configuration.
+	 *
+	 * @since 2.9.5
```
Comment (zackkatz) [inline future/includes/class-gv-renderer.php:145]: No such. Update version to 2.10

**Comment by zackkatz** [comment]
- [ ] Vlad screenshot shows missing field type

**Comment by zackkatz** [comment]
- [ ] [Failing test](https://app.circleci.com/pipelines/github/gravityview/GravityView/549/workflows/5866ecf9-229f-40af-b559-1c2cc29bc0fa/jobs/1811). @mrcasual could this be related to the `gravityview_date_created_adjust_timezone` default change in 2.9.3?

**Comment by zackkatz** [comment]
- [ ] $_GET args not being passed to delete link URLs

**Comment by rafaehlers** [comment]
- [ ] Display issues with Advanced Filter section in Settings panel
image

**Comment by zackkatz** [comment]
- [ ] Fix `?` tooltip icon when using GF 2.5
- [ ] Confirm behavior for layout previews with different license levels and plugin install levels. Ask me what I mean if it's not me doing this!

**Pull request merged**